contracts/mainnet/connectors/mstable/main.sol

```diff
@@ -0,0 +1,363 @@
+pragma solidity ^0.7.6;
+
+/**
+ * @title mStable SAVE.
+ * @dev Depositing and withdrawing directly to Save
+ */
+
+import { Helpers } from "./helpers.sol";
+import { Events } from "./events.sol";
+import { IMasset, ISavingsContractV2, IBoostedSavingsVault, IFeederPool } from "./interface.sol";
+import { TokenInterface } from "../../common/interfaces.sol";
+
+abstract contract mStableResolver is Events, Helpers {
+	//
+	/***************************************
+                    CORE
+    ****************************************/
+
+	/**
+	 * @dev Deposit to Save via mUSD or bAsset
+	 * @notice Deposits token supported by mStable to Save
+	 * @param _token Address of token to deposit
+	 * @param _amount Amount of token to deposit
+	 * @param _minOut Minimum amount of token to mint/deposit, equal to _amount if mUSD
+	 * @return _eventName Event name
+	 * @return _eventParam Event parameters
+	 */
+
+	function deposit(
+		address _token,
+		uint256 _amount,
+		uint256 _minOut
+	) external returns (string memory _eventName, bytes memory _eventParam) {
+		//
+		uint256 mintedAmount = _amount;
+
+		// Check if needs to be minted first
+		if (IMasset(mUsdToken).bAssetIndexes(_token) != 0) {
+			// mint first
+			approve(TokenInterface(_token), mUsdToken, _amount);
+			mintedAmount = IMasset(mUsdToken).mint(
+				_token,
+				_amount,
+				_minOut,
+				address(this)
+			);
+		} else {
+			require(mintedAmount >= _minOut, "mintedAmount < _minOut");
+		}
+
```

Comment:
Hey I think this should emit the event 
```solidity
event LogDeposit(address token, uint256 amount, address path);
```

Reply:
the `_deposit` function returns the event data, does it work? The path is to distinguish the event from where it was deposited:

- path feeder pool address => deposited from feeder pool
- path mUSD address => minted mUSD first
- path imUSD address => just deposited in imUSD

Does it make sense or it is unnecessary?

Reply:
Make sense, just make sure the event data is proper.